fix: LEAP-1442: Fix unexpected scrolling on region select in Video

hlomzik opened the pull request:
When user clicks on region label in Video timeline (left sidepanel of it), the video timeline scrolls to the left.
That happens because click is treated as still a click on timeline but with negative offset outside of scrollable area, so it jumps to previous window.

Also scroll to the first frame of the region on select it in region list. That worked for VideoRectangle, so here is the same code for TimelineRegion. `@todo` is the same, because it's out of the current scope.

hlomzik committed:
fix: LEAP-1442: Fix unexpected scrolling on region select in Video

When user clicks on region label in Video timeline (left sidepanel of it),
the video timeline scrolls to the left.
That happens because click is treated as still a click on timeline but with
negative offset outside of scrollable area, so it jumps to previous window.

hlomzik committed:
Experiment to scroll to the beginning of the region

Let's try to scroll to the first frame of the region on select.

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14517233243)
> Successfully merged:  create mode 100644 web/libs/ui/src/utils/utils.ts

hlomzik committed:
Merge branch 'develop' into 'fb-leap-1442/video-scroll'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14517233243

hlomzik committed:
Don't scroll at all when user clicks everywhere on timeline

Only scroll when user selects region in Outliner

hlomzik committed:
Merge branch 'fb-leap-1442/video-scroll' of github.com:HumanSignal/label-studio into fb-leap-1442/video-scroll

hlomzik committed:
Merge branch 'develop' into fb-leap-1442/video-scroll

hlomzik committed:
Revert some change we don't actually need

vladimirheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14611880874)

vladimirheartex committed:
Merge branch 'develop' into 'fb-leap-1442/video-scroll'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14611880874

vladimirheartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14611880874

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14615478828)
> Successfully merged: Already up to date.

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14621388876)
> Successfully merged:  5 files changed, 19 insertions(+), 5 deletions(-)

hlomzik committed:
Merge branch 'develop' into 'fb-leap-1442/video-scroll'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14621388876

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14621464457)
> Successfully merged: Already up to date.

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14638010941)
> Successfully merged:  18 files changed, 128 insertions(+), 68 deletions(-)

hlomzik committed:
Merge branch 'develop' into 'fb-leap-1442/video-scroll'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14638010941
